### PR TITLE
Make JujuStorage.notices respect the api :-)

### PR DESCRIPTION
**Pull request by chipaca**
Without this change, JujuStorage's notices method would always check
that the notices it yields matched the `event_path` parameter, despite
the docstring saying it was optional.

This fixes that, and adds test for the documented behaviour.

All hail @davigar15 for spotting the bug that resulted from this.

Fixes: #438

### CODE REVIEW

File: ops/storage.py [hunk 1]
```diff
@@ -252,7 +252,7 @@ def notices(self, event_path: str):
         """
         notice_list = self._load_notice_list()
         for row in notice_list:
-            if row[0] != event_path:
+            if event_path and row[0] != event_path:
```
Comment (jameinel) [inline ops/storage.py:255]: This also does weird things for `event_path=""`
Should this be
```
if event_path is not None and ...
```

**Review by jameinel — APPROVED** [review]

File: ops/storage.py (hunk 1 above)
Comment (chipaca) [reply ops/storage.py:255]: that was my first implementation, but the docstring does say `If not supplied (or None/'')`.

Happy to change it to just be None if that's OK. We obviously have no tests either way :-)

**Comment by jameinel** [comment]
If the docstring says we support '' then we can, but we should add a test
to that effect :)

John
=:->

**Comment by chipaca** [comment]
> we should add a test to that effect :)

it's all there, testing nothing, explicit `None`, and `''`.

File: test/test_storage.py [hunk 2]
```diff
@@ -169,6 +169,22 @@ def test_save_notice(self):
...
+        notices = [('e1', 'o1', 'm1'), ('e1', 'o2', 'm2'), ('e2', 'o3', 'm3')]
+        store = self.create_storage()
+        for notice in notices:
+            store.save_notice(*notice)
+
+        for arg, expected in [
+                ('e1', notices[:2]),
+                ('e2', notices[2:]),
+                (None, notices),
+                ('', notices),
```
Comment (chipaca) [inline test/test_storage.py:182]: this is the `notices('')` case. Should I make it more explicit? Probably more readable even if it's a lot of (one-line) repetition

File: test/test_storage.py [hunk 3]
```diff
@@ -169,6 +169,22 @@ def test_save_notice(self):
...
+                ]:
```
Comment (facundobatista) [inline test/test_storage.py:183]: Would you please add, for completeness, a case here `('eX', [])`? thanks!

File: ops/storage.py (hunk 1 above)
Comment (facundobatista) [reply ops/storage.py:255]: Checking for the boolness of event_path is what the SQLite storage was doing, so I think it's better to leave it like that.

**Review by facundobatista — APPROVED** [review]
Awesome, thanks!

File: ops/storage.py (hunk 1 above)
Comment (chipaca) [reply ops/storage.py:255]: ah, for sure if I changed this one I'd have to change the other one (and their docstrings)

File: test/test_storage.py (hunk 3 above)
Comment (chipaca) [reply test/test_storage.py:183]: yarp

**Pull request merged**